trie.go

```diff
@@ -38,6 +38,31 @@ func New() *Trie {
 	}
 }
 
+// Free all Nodes of the Trie, renew the root Node.
+func Free(t *Trie) {
```

Comment:
Sorry for being so late to review this. Could you provide a use case in the PR message and also this needs a test.

Aside from that, I'd prefer this to be a method on `*Trie` and be named `(*Trie).Reset()`.